### PR TITLE
Extended Dataset type config to provide search options like strategy

### PR DESCRIPTION
**Pull request by amsiglan**
### Description
Extended Dataset type config to provide search options like strategy to shift the strategy provided by language to the dataset itself.

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/query_enhancements/public/search/sql_search_interceptor.ts [hunk 1]
```diff
@@ -39,7 +39,7 @@ export class SQLSearchInterceptor extends SearchInterceptor {
     const isAsync = strategy === SEARCH_STRATEGY.SQL_ASYNC;
```
Comment (kavilla) [inline src/plugins/query_enhancements/public/search/sql_search_interceptor.ts:39]: we can get rid of this correct?

File: src/plugins/query_enhancements/public/search/sql_search_interceptor.ts [hunk 2]
```diff
@@ -54,7 +54,18 @@ export class SQLSearchInterceptor extends SearchInterceptor {
 
   public search(request: IOpenSearchDashboardsSearchRequest, options: ISearchOptions) {
     const dataset = this.queryService.queryString.getQuery().dataset;
-    const strategy = dataset?.type === DATASET.S3 ? SEARCH_STRATEGY.SQL_ASYNC : SEARCH_STRATEGY.SQL;
+    const datasetType = dataset?.type;
+    let datasetTypeConfig;
+
+    if (datasetType) {
```
Comment (kavilla) [inline src/plugins/query_enhancements/public/search/sql_search_interceptor.ts:60]: nit:

would it better to have
```
let strategy;
```

File: src/plugins/query_enhancements/public/search/sql_search_interceptor.ts (hunk 1 above)
Comment (amsiglan) [reply src/plugins/query_enhancements/public/search/sql_search_interceptor.ts:39]: yeah

File: src/plugins/query_enhancements/public/search/sql_search_interceptor.ts (hunk 2 above)
Comment (amsiglan) [reply src/plugins/query_enhancements/public/search/sql_search_interceptor.ts:60]: refactored

**Review by kavilla — APPROVED** [review]

**Review by ashwin-pc — APPROVED** [review]

**Pull request merged**